forch/mab_state_machine.py

```diff
@@ -0,0 +1,159 @@
+"""Module that implements MAB state machine"""
+
+import logging
+from threading import Timer
+
+LOGGER = logging.getLogger('mabsm')
+
+
+class State():
```

Comment:
way too complex. I don't understand why all this is necessary for 4 states that have only a handful of transitions. This level of abstraction really isn't needed.  Please see the faucet of daq state machines for simple ways of handling this.  It really can/should just be one simple variable that says "this is my current state". When you want to change it based off of some event, you simply check the current state, and then set the new state appropriate, or log a warning.  Maybe create an intermediate helper method _state_transition() -- ala DAQ.